Automattic/wp-calypso
# Subscriptions: Add the Subscribe block post end placement toggle to the Newsletter Settings

Related to https://github.com/Automattic/wp-calypso/issues/88462

## Proposed Changes

It:
- adds the Subscribe block post end placement toggle to the Subscriptions Settings (behind the `jetpack/subscription-site` feature flag, disabled on prod)
- adds some basic tracking to it
- changes structure of the Subscriptions Settings to mimic what we have in Jetpack 

### Before

<img width="734" alt="Screenshot 2024-03-15 at 11 12 36" src="https://github.com/Automattic/wp-calypso/assets/4068554/bf58a3d9-2a1d-4626-a189-a07c11c0b199">

### After

<img width="734" alt="Screenshot 2024-03-15 at 11 13 17" src="https://github.com/Automattic/wp-calypso/assets/4068554/d25393ac-aca3-4bca-8a1e-63627d2ae35a">

<img width="734" alt="Screenshot 2024-03-15 at 11 13 23" src="https://github.com/Automattic/wp-calypso/assets/4068554/0493f81a-0f40-48da-b741-5207547da888">

<img width="734" alt="Screenshot 2024-03-15 at 11 13 27" src="https://github.com/Automattic/wp-calypso/assets/4068554/eac1afee-984d-43cf-b1d6-3b5cf25cf088">

<br/>
<br/>

**Note to myself:** Add the "String Freeze" label before the merge!

## Testing Instructions



1. Go to the Newsletter Settings
2. Make sure the changes to the Subscribe settings section look sane

## Pre-merge Checklist
